torchvision/transforms/functional_tensor.py

```diff
@@ -236,3 +276,51 @@ def _blend(img1, img2, ratio):
     # type: (Tensor, Tensor, float) -> Tensor
     bound = 1 if img1.dtype in [torch.half, torch.float32, torch.float64] else 255
     return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+
+
+def _rgb2hsv(img):
+    r, g, b = img[0], img[1], img[2]
+
+    maxc, _ = torch.max(img, dim=0)
+    minc, _ = torch.min(img, dim=0)
+    uv = maxc
+
+    cr = maxc.to(dtype=torch.float32) - minc.to(dtype=torch.float32)
+    s = cr / maxc.to(dtype=torch.float32)
+    rc = (maxc-r).to(dtype=torch.float32) / cr
+    gc = (maxc-g).to(dtype=torch.float32) / cr
+    bc = (maxc-b).to(dtype=torch.float32) / cr
+
+    s = (maxc != minc) * s
+    hr = (maxc == r) * (bc - gc)
+    hg = ((maxc == g) & (maxc != r)) * (2.0 + rc - bc)
+    hb = ((maxc != g) & (maxc != r)) * (4.0 + gc - rc)
+    h = (hr + hg + hb)
+
+    #  torch.fmod and  math.fmod have different precision.
+    h = torch.fmod((h / 6.0 + 1.0), 1.0)
+
+    if img.dtype == torch.uint8:
+        uh = torch.clamp((h*255.0).to(dtype=torch.int32), 0, 255).to(dtype=torch.uint8)
+        us = torch.clamp((s*255.0).to(dtype=torch.int32), 0, 255).to(dtype=torch.uint8)
+    else:
+        uh = torch.clamp(h * 255.0, 0.0, 255.0)
+        us = torch.clamp(s * 255.0, 0.0, 255.0)
```

Comment:
Given that this function normalizes by the max value, it might make sense to expect the tensor to be fed to it to be a float tensor, and to always return float tensors in 0-1 for this helper function, and then use the (yet not merged) `convert_image_dtype` https://github.com/pytorch/vision/pull/2078 to cast from float to the original dtype.

Reply:
This function normalizes h by maxc-minc, s by maxc and v is not normalized. I dont completelt get your point here. I can change this function to accept [0, 1]. But I am wondering if we keep it the same as in PIL will it be better?

Reply:
So, here is what I meant (and I missed that `v` was not invariant, sorry about that):
- both `h` and `s` are invariant by scaling, so `_rgb2hsv(img)[0:2] == _rgb2hsv(img * 255)[0:2]` (minus floating point errors)
- `v` is equivariant by scaling, so `_rgb2hsv(img)[2] == _rgb2hsv(img * 255)[2] / 255`

This means that the range of the input doesn't matter for `h` and `s`, and the range for `v` is just scaled according to the max value.

This means that, for simplicity, we can remove the `* 255` in this part of the function (which is specific to `uint8` representation), and instead scale the whole output tensor by the `convert_image_dtype` (or equivalent) in the main function, as it would make things simpler (and could potentially support other image type representations)

Reply:
Done.